### PR TITLE
Add search-collector lease addon on managed clusters

### PR DESCRIPTION
**Pull request by SherinV**
**Related Issue:**  open-cluster-management/backlog#6473

### Description of changes
Create and update search-collector lease on managed cluster instead of remote cluster namespace in hub.
Made necessary permission updates in klusterlet-addon-operator search chart. https://github.com/open-cluster-management/klusterlet-addon-operator/pull/21
Add test

### CODE REVIEW

File: pkg/config/kubeClient.go [hunk 1]
```diff
@@ -53,9 +54,18 @@ func getKubeConfig() *rest.Config {
...
 func GetDiscoveryClient() *discovery.DiscoveryClient {
-	discoveryClient, err := discovery.NewDiscoveryClientForConfig(getKubeConfig())
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(GetKubeConfig())
 	if err != nil {
 		glog.Fatal("Cannot Construct Discovery Client From Config: ", err)
 	}
 	return discoveryClient
 }
+
+func GetKubeClient() *kubernetes.Clientset {
```
Comment (mqhackett) [inline pkg/config/kubeClient.go:64]: This was previously used code? Commented out and saved for later? It's finally later! Good move, we need this method now.

File: pkg/lease/lease.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		} else {
+			glog.Infof("addon lease %q/%q on local managed cluster updated", r.componentNamespace, r.LeaseName)
+		}
+		glog.Info("Exiting lease Reconcile")
+
+		return
+	}
+}
+
+func getPodNamespace() string {
```
Comment (mqhackett) [inline pkg/lease/lease.go:75]: The value of creating a separate package in our project is exemplified by this method. While it was previously recommended to put the lease creation logic in our operator, our current implementation disqualifies this option. It certainly would be possible, but would create a spaghetti code monster. 

By creating a separate function, we achieve functionality exposed by the operators that does not exist at the search-collector. Example, in the operator, we could just call a function that gives us the namespace, which is required for creating the lease. However, here, we need this method. 

Very clean way of doing this. I'd consider a name other than "lease" to avoid future confusion, but tbh i'm happy with 'lease' for now as it shouldn't cause any code-breaking collisions.

File: pkg/config/kubeClient.go [hunk 3]
```diff
@@ -32,7 +33,7 @@ func GetDynamicClient() dynamic.Interface {
 	return dynamicClient
 }
 
-func getKubeConfig() *rest.Config {
+func GetKubeConfig() *rest.Config {
```
Comment (jlpadilla) [inline pkg/config/kubeClient.go:36]: Is this change needed? I don't see `GetKubeConfig()` being used from another package.

File: pkg/lease/lease.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+				Namespace: r.componentNamespace,
+			},
+			Spec: coordinationv1.LeaseSpec{
+				LeaseDurationSeconds: &r.LeaseDurationSeconds,
+				RenewTime: &metav1.MicroTime{
+					Time: time.Now(),
+				},
+			},
+		}
+		if _, err := r.KubeClient.CoordinationV1().Leases(r.componentNamespace).Create(lease); err != nil {
```
Comment (jlpadilla) [inline pkg/lease/lease.go:43]: Most likely we'll need to update the pod permissions to do this create action.

Comment (jlpadilla) [reply pkg/lease/lease.go:43]: And the update action below.

**Review by jlpadilla — APPROVED** [review]
Left a couple minor comments.
/lgtm
/hold

File: pkg/lease/lease.go (hunk 4 above)
Comment (SherinV) [reply pkg/lease/lease.go:43]: We updated the pod permissions for managed cluster collector pods in https://github.com/open-cluster-management/klusterlet-addon-operator/pull/21/files. Do we need to update the search-chart too? @jlpadilla

File: pkg/config/kubeClient.go (hunk 3 above)
Comment (SherinV) [reply pkg/config/kubeClient.go]: Updated.

File: pkg/lease/lease.go (hunk 4 above)
Comment (jlpadilla) [reply pkg/lease/lease.go:43]: @SherinV that's all we needed. Thanks for confirming! 👍

Comment (mqhackett) [reply pkg/lease/lease.go:43]: Could you explain why we needed to update the pod permissions for this call?

**Review by mqhackett — APPROVED** [review]

File: pkg/lease/lease.go (hunk 4 above)
Comment (SherinV) [reply pkg/lease/lease.go:43]: So that the pod has create and update permissions on lease objects. Otherwise the create will fail with `pod doesn't have permission to create/update lease in this namespace`
